app/models/theme.rb

```diff
@@ -893,8 +894,9 @@ def migrate_settings(start_transaction: true)
             name: res[:name],
             theme_field_id: res[:theme_field_id],
           )
+
         record.calculate_diff(res[:settings_before], res[:settings_after])
-        record.save!
+        allow_out_of_sequence_migration ? record.save : record.save!
```

Comment:
I don't get this change, can you explain why we need to use `.save` when `allow_out_of_sequence_migration`?

Reply:
I had a closer look and I think this is the best change because still want to create the `theme_settings_migration` records when validation does not fail. I added a comment to explain why we use `save` instead of `save!` though. Can you see if that is clearer?

 ```
 # If out of sequence migration is allowed we don't want to raise an error if the record is invalid due to version
 # conflicts
 ```